### PR TITLE
Add transaction submission api test

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
Need https://github.com/aptos-labs/aptos-ts-sdk/pull/65 to land first.

Add transaction submission api function tests

secp256k1 is skipped for now because of a user txn processor issue https://github.com/aptos-labs/aptos-ts-sdk/pull/42

* submit single signer transaction with script / entry function / multisig payloads
* submit multi agent transaction with script / entry function payloads
* submit fee payer transaction with script / entry function / multisig payloads

### Test Plan
pnpm test

### CODE REVIEW

**Comment by xbtmatt** [comment]
At some point we should probably try to use the same little "test harness" for these lol, there's a ton of setup

File: tests/e2e/transaction/transaction_submission.test.ts [hunk 1]
```diff
@@ -3,100 +3,362 @@
...
+        const response = await aptos.submitTransaction({
+          transaction,
+          senderAuthenticator,
+          secondarySignerAuthenticators: { feePayerAuthenticator: feePayerSignerAuthenticator },
         });
-        const bobauthenticator = aptos.signTransaction({
-          signer: bob,
-          transaction: rawTxn,
+
+        await waitForTransaction({ aptosConfig: config, txnHash: response.hash });
```
Comment (xbtmatt) [inline tests/e2e/transaction/transaction_submission.test.ts:139]: I think in general we should be doing a check here right?

The node api will tell you if you have a fee payer  signature, etc

```typescript
const responseSignature = response.signature as TransactionFeePayerSignature;
expect(responseSignature.type).toEqual("fee_payer_signature");
```

this is what I do in mine

the 5 types:

```
ed25519_signature
secp256k1_ecdsa_signature
multi_agent_signature
fee_payer_signature
multi_ed25519_signature
```

File: tests/e2e/transaction/transaction_submission.test.ts [hunk 2]
```diff
@@ -3,100 +3,362 @@
...
+  const senderAccount = Account.generate();
+  const recieverAccounts = [Account.generate(), Account.generate()];
+  const senderSecp256k1Account = Account.generate(SigningScheme.Secp256k1Ecdsa);
+  const secondarySignerAccount = Account.generate();
+  const feePayerAccount = Account.generate();
+  beforeAll(async () => {
+    await fundAccounts(aptos, [
+      senderAccount,
+      senderSecp256k1Account,
+      ...recieverAccounts,
```
Comment (xbtmatt) [inline tests/e2e/transaction/transaction_submission.test.ts:21]:
```suggestion
      ...receiverAccounts,
```

ctrl+f `reciever` nit :D

File: tests/e2e/transaction/transaction_submission.test.ts (hunk 1 above)
Comment (0xmaayan) [reply tests/e2e/transaction/transaction_submission.test.ts:139]: hmm I guess it depends on what you are testing. those tests tests that transaction is submitted. We have some other tests that tests the signature in `sign_transaction`

**Pull request closed without merging**